internal/commands/root/root.go

```diff
@@ -156,7 +156,7 @@ func runRootCommandWithProviderAndClient(ctx context.Context, pInit provider.Ini
 
 	pNode := NodeFromProvider(ctx, c.NodeName, taint, p, c.Version)
 	nodeRunner, err := node.NewNodeController(
-		node.NaiveNodeProvider{},
+		p,
```

Comment:
WDYT about checking if `p` implements `NodeProvider` and using it if so, with a fallback to naive?